dlc-messages/src/ser_impls.rs

```diff
@@ -537,6 +538,23 @@ pub fn read_i32<R: ::std::io::Read>(reader: &mut R) -> Result<i32, DecodeError>
     Ok(i32::from_be_bytes(v))
 }
 
+/// Writes an `i64` value to the given writer.
+pub fn write_i64<W: Writer>(i: &i64, writer: &mut W) -> Result<(), ::std::io::Error> {
+    let i = i.to_be_bytes();
+    let i1: [u8; 4] = i[0..4].try_into().unwrap();
+    let i2: [u8; 4] = i[4..8].try_into().unwrap();
+    i1.write(writer)?;
+    i2.write(writer)
+}
+
+/// Reads an `i64` value from the given reader.
+pub fn read_i64<R: ::std::io::Read>(reader: &mut R) -> Result<i64, DecodeError> {
+    let v1: [u8; 4] = Readable::read(reader)?;
+    let v2: [u8; 4] = Readable::read(reader)?;
+    let v = [v1, v2].concat();
+    Ok(i64::from_be_bytes(v.try_into().unwrap()))
+}
```

Comment:
Because pnl is i64, i64 needs to be readable / writable. I tried to mirror your implementation for i32, but it seems that Readable/Writeable in rust-lightning are not implemented for [u8; 8].  im not a huge fan of line 554 because i believe concat creates a heap allocation by going through a Vec first.

with that being said, on the nightly compiler, the concat method can simply generate an array without going through vec.

https://doc.rust-lang.org/std/slice/trait.Concat.html